src/js/tracker.js

```diff
@@ -916,6 +923,23 @@
 		}
 
 		/**
+		 * Check that *both* optimizely and optimizely.data exist
+		 *
+		 * @param property optimizely data property
+		 * @param snd optional nested property
+		 */
+		function getOptimizelyXData(property, snd) {
+			var data;
+			if (windowAlias.optimizely && windowAlias.optimizely.data) {
+				data = windowAlias.optimizely.get(property);
+				if (typeof snd !== 'undefined' && data !== undefined) {
+					data = data[snd]
+				}
+			}
+			return data
+		}
```

Comment:
If we're talking about `getOptimizelyData`, `windowAlias.optimizely.data` must not be deleted. This is a defensive check fixing https://github.com/snowplow/snowplow-javascript-tracker/issues/553. That bug can result in whole frontend freeze - if user defined somewhere empty `window.optimizely` object - our tracker will try access to  `undefined` (as `optimizely.data` is `undefined`) which will throw exception freezing whole execution thread.